backend/src/main/java/com/github/JeTSkY1h/book/BookService.java

```diff
@@ -1,79 +1,75 @@
     EpubReader epubReader = new EpubReader();
-
-    public BookService(BookRepo bookRepo) {
-        this.bookRepo = bookRepo;
-    }
-
     public Optional<Book> getById(String id) {
         return bookRepo.findById(id);
     }
-
     public List<Book> getBooks() {
         return bookRepo.findAll();
     }
 
     public List<Book> refresh() {
-        List<File> books;
-        File f = new File(path);
-        //Todo Check in DB for dubles
-        books = Arrays.stream(f.listFiles()).filter(bookFile -> bookFile.getAbsolutePath().endsWith(".epub")).toList();
+
+        List<URL> bookURLs = cloudinaryService.getBookURLs();
+
         List<Book> res = new ArrayList<>();
-        for (File bookFile : books) {
-            try (FileInputStream fIn = new FileInputStream(bookFile.getPath())) {
-                nl.siegmann.epublib.domain.Book book = epubReader.readEpub(fIn);
-                Book bookRes = new Book();
-                BufferedImage buffCoverImg = ImageIO.read(book.getCoverImage().getInputStream());
-                File outputFile = new File(path + "/" + book.getTitle().replaceAll("[-+/\\.:'*;, ]", "") + ".png");
-                ImageIO.write(buffCoverImg, "png", outputFile);
-                bookRes.setCoverPath(outputFile.getAbsolutePath());
-                bookRes.setAuthor(book.getMetadata().getAuthors().get(0).toString());
-                bookRes.setFilePath(bookFile.getPath());
-                bookRes.setGenre(book.getMetadata().getSubjects());
-                bookRes.setTitle(book.getTitle());
-                System.out.println(book.getMetadata().getDescriptions());
-                if(book.getMetadata().getDescriptions().size() > 0) {
-                    bookRes.setDescription(book.getMetadata().getDescriptions().get(0));
+
+            bookURLs.forEach(bookFile -> {
+                System.out.println("Test"+  bookFile.getPath());
+                try (InputStream fIn = bookFile.openStream()) {
+                    nl.siegmann.epublib.domain.Book book = epubReader.readEpub(fIn);
+                    Book bookRes = new Book();
+                    Map imageUploadRes = cloudinaryService.uploadCover(book.getCoverImage().getData());
+                    bookRes.setCoverPath((String) imageUploadRes.get("secure_url"));
+                    bookRes.setAuthor(book.getMetadata().getAuthors().get(0).toString());
+                    bookRes.setFilePath(bookFile);
+                    bookRes.setGenre(book.getMetadata().getSubjects());
+                    bookRes.setTitle(book.getTitle());
+                    System.out.println(book.getMetadata().getDescriptions());
+                    if(book.getMetadata().getDescriptions().size() > 0) {
+                        bookRes.setDescription(book.getMetadata().getDescriptions().get(0));
+                    }
+                    res.add(bookRes);
+                    bookRepo.saveAll(res);
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
Instead of `e.printStackTrace()` and `System.out.println(..)` I would prefer to use slf4j for logging.